language/stdlib/modules/0L/Wallet.move

```diff
@@ -0,0 +1,411 @@
+          is_frozen: false,
+          consecutive_rejections: 0,
+          unfreeze_votes: Vector::empty<address>()
+        })
+      }
+    }
+
+
+    // Utility for vm to set an address as a Community Wallet
+    public fun vm_set_comm(vm: &signer, addr: address) acquires CommunityWallets {
+      CoreAddresses::assert_libra_root(vm);
+      if (exists<CommunityWallets>(0x0)) {
+        let list = get_comm_list();
+        if (!Vector::contains<address>(&list, &addr)) {
+        
+          let s = borrow_global_mut<CommunityWallets>(0x0);
+          Vector::push_back(&mut s.list, addr);
+        }
+      }
+    }
+
+    // Utility for vm to remove the CommunityWallet tag from an address
+    public fun vm_remove_comm(vm: &signer, addr: address) acquires CommunityWallets {
+      CoreAddresses::assert_libra_root(vm);
+      if (exists<CommunityWallets>(0x0)) {
+        let list = get_comm_list();
+        let (yes, i) = Vector::index_of<address>(&list, &addr);
+        if (yes) {
+          let s = borrow_global_mut<CommunityWallets>(0x0);
+          Vector::remove(&mut s.list, i);
+        }
+      }
+    }
+
+
+  // The community wallet Signer can propose a timed transaction.
+  // the timed transaction defaults to occurring in the 3rd following epoch.
+  // TODO: Increase this time?
+  // the transaction will automatically occur at the epoch boundary, unless a veto vote by the validator set is successful.
+  // at that point the transaction leves the proposed queue, and is added the rejected list.
+  public fun new_timed_transfer(sender: &signer, payee: address, value: u64, description: vector<u8>): u64 acquires CommunityTransfers, CommunityWallets {
+      let sender_addr = Signer::address_of(sender);
+      let list = get_comm_list();
+        
+      assert(
+        Vector::contains<address>(&list, &sender_addr), Errors::requires_role(ERR_PREFIX + 001)
+      );
+
+      let d = borrow_global_mut<CommunityTransfers>(0x0);
+      d.max_uid = d.max_uid + 1;
+      
+      // add current epoch + 1
+      let current_epoch = LibraConfig::get_current_epoch();
+
+      let t = TimedTransfer {
+          uid: d.max_uid,
+          expire_epoch: current_epoch + 3,
+          payer: sender_addr,
+          payee: payee,
+          value: value,
+          description: description,
+          veto: Veto {
+            list: Vector::empty<address>(),
+            count: 0,
+            threshold: 0,
+          }
+      };
+
+      Vector::push_back<TimedTransfer>(&mut d.proposed, t);
+      return d.max_uid
+    }
+
+  // utlity to query a CommunityWallet transfer wallet.
+  // Note: doesn not need to be a public function, except for use in tests.
+  public fun find(uid: u64, type_of: u8): (Option<TimedTransfer>, u64) acquires CommunityTransfers {
+    let c = borrow_global<CommunityTransfers>(0x0);
+    let list = if (type_of == 0) {
+      &c.proposed
+    } else if (type_of == 1) {
+      &c.approved
+    } else {
+      &c.rejected
+    };
```

Comment:
Should there be another option to search all three? What if I want to know whether it has been approved or rejected? I won't know which list to search.

Reply:
Good idea.

Reply:
Will add as a new issue.